File: 2021/Go/main.go
package main

import (
	"fmt"
	"strconv"

	"github.com/herulume/Advent-of-Code/pkg/day1"
)

func main() {
	d1p1, err := day1.Day1Part1()
	if err != nil {
		fmt.Print(err)
	}

	d1p2, err := day1.Day1Part2()
	if err != nil {
		fmt.Print(err)
	}

	fmt.Println("Day 1 solutions")
	fmt.Println("Part 1: " + strconv.Itoa(d1p1))
	fmt.Println("Part 2: " + strconv.Itoa(d1p2))
}

File: 2021/Haskell/app/Main.hs
module Main where

import qualified Day1
import qualified Day2

day1 :: [IO ()]
day1 = [ putStrLn "Day 1 solutions"
       , putStr "Part 1: " >> Day1.p1
       , putStr "Part 2: " >> Day1.p2
       ]

day2 :: [IO ()]
day2 = [ putStrLn "Day 2 solutions"
       , putStr "Part 1: " >> Day2.p1
       , putStr "Part 2: " >> Day2.p2
       ]
       

main :: IO ()
main = do
  sequence_ day1
  sequence_ day2

